packages/openapi-generator/src/parser/media-type.ts

```diff
@@ -0,0 +1,85 @@
+import { createLogger } from '@sap-cloud-sdk/util';
+import { OpenAPIV3 } from 'openapi-types';
+import { OpenApiSchema } from '../openapi-types';
+import { parseSchema } from './schema';
+
+const logger = createLogger('openapi-generator');
+/**
+ * Parse the type of a resolved request body or response object.
+ * @param bodyOrResponseObject The request body or response object to parse the type from.
+ * @returns The type name of the request body if there is one.
+ */
+export function parseApplicationJsonMediaType(
+  bodyOrResponseObject:
+    | OpenAPIV3.RequestBodyObject
+    | OpenAPIV3.ResponseObject
+    | undefined
+): OpenApiSchema | undefined {
+  if (bodyOrResponseObject) {
+    const mediaType = getMediaTypeObject(
+      bodyOrResponseObject,
+      'application/json'
+    );
+    const schema = mediaType?.schema;
+    if (schema) {
+      return parseSchema(schema);
+    }
+  }
+}
+
+export function parseMediaType(
+  bodyOrResponseObject:
+    | OpenAPIV3.RequestBodyObject
+    | OpenAPIV3.ResponseObject
+    | undefined
+): OpenApiSchema | undefined {
+  const allMediaTypes = getMediaTypes(bodyOrResponseObject);
+  if (allMediaTypes.length) {
+    const jsonMediaType = parseApplicationJsonMediaType(bodyOrResponseObject);
+
+    if (!jsonMediaType) {
```

Comment:
I'm afraid log every occurrence might be too much.

Reply:
True, should we do this as debug only? This is the same for the type parsing btw, I would do the same there.

Reply:
Done.